shakedown/dcos/__init__.py

```diff
@@ -162,3 +260,86 @@ def _gen_url(url_path):
     """
     from six.moves import urllib
     return urllib.parse.urljoin(dcos_url(), url_path)
+
+
+# Initialize some things.
+transportManager = TransportManager()
+
+
+class HostSession:
+    """Context manager that returns an SSH session to run commands.
+    
+    """
+    def __init__(self, host, username, key_path, verbose):
+        self.host = host
+        self.username = username
+        self.key_path = key_path
+        self.verbose = verbose
+        self.exit_code = -1
+        self.output = ''
+        self.session = None
+    
+    def __enter__(self):
+        """Called when using a `with` closure.
+
+        :return: this session manager
+        :rtype: HostSession
+        """
+        self.session = transportManager.get_session(
+                self.host,
+                self.username,
+                self.key_path)
+        return self
+    
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """Executed when the context manager is complete (exits
+        the with closure).
+
+        :param exc_type:
+        :param exc_val:
+        :param exc_tb:
+        :return: None
+        """
+        self.exit_code = self.session.recv_exit_status()
+        self._wait_for_recv()
+        # read data that is ready
+        while self.session.recv_ready():
+            rl, wl, xl = select.select([self.session], [], [], 0.0)
```

Comment:
this is some of the more complex code...  and the variables are not clear

Reply:
This was copied from `command.py`. I will see what I can do to update these names.